radixdlt/babylon-wallet-android
# Refactor ActionableAddressView to make it work in previews

## Description
There are several reasons that some previews don't work. But those that cause the most problems are:
1. the RET in the `Resource` domain model
2. `ActionableAddressView` that uses Hilt

The focus of this PR was mostly on the 2. in order to get previews for account cards and the wallet screen. That was achieved by doing two things:

 ➡️    Applied a facade pattern for the Hilt interface `ActionableAddressViewEntryPointI` and now we can use the `ActionableAddressViewEntryPointMock` for previews and the actual `ActionableAddressViewEntryPoint` for the real UI.

➡️    Used the `CompositionLocalProvider` in `MainActivity` to provide the actual `ActionableAddressViewEntryPoint` at runtime. The initial idea was to pass the mock in the `RadixWalletTheme` of the previews but that doesn't work because Hilt `EntryPoint` needs the application context. See the file `ActionableAddressViewEntryPointI.kt`. 

About 1. I just replaced the accounts and assets with empty lists of resources. At least with this way we can see several screens like WalletScreen which doesn't only provides information about assets, but also account warnings and total balance!

## How to test

1. Build the app and see the previews
2. Run the app and play with the ActionableAddressView

## Screenshot

<img width="200" alt="Screenshot 2024-02-25 at 7 38 22 PM" src="https://github.com/radixdlt/babylon-wallet-android/assets/118305718/fa2befe7-d928-45fc-a73b-1dea5cc3105e">


## PR submission checklist
- [X] I have tested the ActionableAddressView in the app
